cirq/devices/device.py

```diff
@@ -46,6 +48,36 @@ def qubit_set(self) -> Optional[AbstractSet['cirq.Qid']]:
         # Default to the qubits being unknown.
         return None
 
+    @property
+    def edges(self) -> Optional[List[Tuple['cirq.Qid', 'cirq.Qid']]]:
```

Comment:
I'd suggest typing this as `Optional[AbstractSet[Tuple['cirq.Qid', 'cirq.Qid']]]`, by analogy with `qubit_set`. In addition, we might consider changing the name to something more specific than "edges", for example `qubit_pairs` or `coupled_pairs`.

Note that in some internal code we have a property like this, but it's a persistent source of bugs that tuples are not symmetric: `(q0, q1) != (q1, q0)`. This can lead to missing coupled pairs if you forget to sort the qubits in a candidate pair before checking for it in the collection returned here. Tuples are certainly convenient, but it might be worth considering whether there is a way to avoid this issue, e.g. use our own "pair" type which is symmetric, or return a custom collection that sorts before checking membership, or just leave as-is but document that pairs here are sorted.

Reply:
Yes, I was thinking of making the pair a frozenset instead of a tuple. So the return type would be optional(frozenset(frozenset(qid))).

Actually, maybe a custom Pair type would be better, a thin wrapper around frozen set that guarantees only two elements. I'll add that before merging.